### PR TITLE
Added basic tracks for "shipping methods added to shipping zones"

### PR DESCRIPTION
**Pull request by DanReyLop**
Related to #15316

@kellychoffman @justinshreve What exactly do we want to track here? It wouldn't be hard to add more detailed tracking, but I didn't want to overdo it. This is what's implemented now:
* Every time a shipping method is **added** to a zone (and the zone is saved), a `tracks` event with the type of the method (`free_shipping`, `local_pickup` or `flat_rate`) is recorded.
* Every time the user changes the type of a shipping method, it's recorded too as if it was a new method creation (since that's what it is internally).
* The event is recorded even if the user saves a `DISABLED` method.

Do we want to track when the method is removed too? And when the method is toggled between enabled/disabled?

### CODE REVIEW

**Comment by kellychoffman** [comment]
> What exactly do we want to track here?

For v1, we agreed to only track ```shipping methods used```

**Comment by DanReyLop** [comment]
Done. All the tracks events have a `shippingMethod` prop with the kind of shipping method they're referring to.

Event types:
* `calypso_woocommerce_shipping_method_created`
* `calypso_woocommerce_shipping_method_deleted`
* `calypso_woocommerce_shipping_method_enabled`
* `calypso_woocommerce_shipping_method_disabled`

Behaviour:
* When you add a shipping method and save: `method_created` + `method_enabled`.
* When you add a shipping method but you disable it before saving: `method_created` + `method_disabled`
* When you enable a disabled method and save: `method_enabled`
* When you disable an enabled method and save: `method_disabled`
* When you remove a shipping method and save: `method_deleted`
* When you change a method type and save: `method_deleted` (with the old type) + `method_created` (with the new type) + `method_enabled/disabled` (depending on if the new method is enabled or disabled).

Is this OK? Am I missing something that will make this data unusable? :)

**Comment by kellychoffman** [comment]
Logic sounds good to me but I'll wait for a second opinion.

File: client/extensions/woocommerce/state/data-layer/ui/shipping-zones/index.js [hunk 1]
```diff
@@ -143,10 +144,15 @@ const getAutoOrderZonesSteps = ( siteId, zoneId, serverZones, allServerLocations
...
 
-const getZoneMethodDeleteSteps = ( siteId, zoneId, method ) => {
+const getZoneMethodDeleteSteps = ( siteId, zoneId, method, state ) => {
+	const methodType = getShippingZoneMethod( state, method.id, siteId ).methodType;
+
 	return [ {
 		description: translate( 'Deleting Shipping Method' ),
 		onStep: ( dispatch, actionList ) => {
+			analytics.tracks.recordEvent( 'calypso_woocommerce_shipping_method_deleted', {
+				shippingMethod: methodType,
```
Comment (justinshreve) [inline client/extensions/woocommerce/state/data-layer/ui/shipping-zones/index.js]: property names need to be lowercase and separated by underscores. While testing this, I noticed that the events are going into the "rejects" table -- and I think the property naming is the reason for that.

https://fieldguide.automattic.com/tracks-naming-conventions/

Comment (justinshreve) [inline client/extensions/woocommerce/state/data-layer/ui/shipping-zones/index.js]: I just noticed I did the exact same thing for payment methods... I'll fix mine in a separate PR :).

Comment (DanReyLop) [inline client/extensions/woocommerce/state/data-layer/ui/shipping-zones/index.js]: > I just noticed I did the exact same thing for payment methods... I'll fix mine in a separate PR :).

Yeah, guess where I copy&pasted from 😂

**Pull request merged**